core/amber/src/main/scala/edu/uci/ics/texera/workflow/common/workflow/PhysicalPlan.scala

```diff
@@ -35,29 +35,14 @@ object PhysicalPlan {
       physicalPlan = physicalPlan.addSubPlan(subPlan)
     })
 
-    // connect inter-operator links
+    // connect external links
     logicalPlan.links.foreach(link => {
-      val fromLogicalOp = link.origin.operatorId
-      val fromPort = link.origin.portOrdinal
-      val fromPortName = logicalPlan.operators
-        .filter(op => op.operatorIdentifier == link.origin.operatorId)
-        .head
-        .operatorInfo
-        .outputPorts(fromPort)
-        .displayName
-      val fromOp = physicalPlan.getPhysicalOpForOutputPort(fromLogicalOp, fromPortName)
-
-      val toLogicalOp = logicalPlan.getOperator(link.destination.operatorId).operatorIdentifier
-      val toPort = link.destination.portOrdinal
-      val toPortName = logicalPlan.operators
-        .filter(op => op.operatorIdentifier == link.destination.operatorId)
-        .head
-        .operatorInfo
-        .inputPorts(toPort)
-        .displayName
-      val toOp = physicalPlan.getPhysicalOpForInputPort(toLogicalOp, toPortName)
-
-      physicalPlan = physicalPlan.addLink(PhysicalLink(fromOp.id, fromPort, toOp.id, toPort))
+
+      val fromOp = physicalPlan.getPhysicalOpForOutputPort(link.fromOpId, link.fromPortId)
```

Comment:
physicalPlan.getOperator(link.fromOpId)?

Reply:
here this is logical link, so we have to conduct such a search for the physical op.